src/controllers/controller.h

```diff
@@ -193,13 +196,33 @@ class ControllerJSProxy : public QObject {
             : m_pController(m_pController) {
     }
 
+    QHash<QString, QString> m_cache;
+
     // The length parameter is here for backwards compatibility for when scripts
     // were required to specify it.
     Q_INVOKABLE virtual void send(const QList<int>& data, unsigned int length = 0) {
         Q_UNUSED(length);
         m_pController->send(data, data.length());
     }
 
+    // Available charsets should be available here:
+    // http://www.iana.org/assignments/character-sets/character-sets.xhtml
```

Comment:
alternatively we just let the mapping handle it. Not sure whats better.

Reply:
Meh. I'm not in favor of that. The idea is to allow controller developers to use charsets if available without having to open a PR to make it available. I can, however check against `QStringConverter::availableCodecs()` and `QTextCodec::availableCodecs()`.

Reply:
Every Mixxx installation must support each mapping. The mapping API must be stable accross installations. There is no problem, to add a long list of charsets to the positive list.

Reply:
How likely is the case that QT is build without ICU? It is a core feature of every Linux distribution and for Window an macOs we provide our own solution. 

In the unlikely case a certain codec is not available we get an empty string. This is mich better than failed mapping. 

So I guess we can be relaxed here and use an easy solution like a simple isEmpt() sanity check or such.

Reply:
> In the unlikely case a certain codec is not available we get an empty string

No. Current implementation returns `undefinedè and I documented that it will return `undefined` if charset is unavailable. This lets mapping developers implement an alternative behavior.